core/modules/flowgen.cc

```diff
@@ -84,13 +84,17 @@ inline uint64_t FlowGen::NextFlowArrival() {
 }
 
 inline struct flow *FlowGen::ScheduleFlow(uint64_t time_ns) {
+  struct flow *f;
+
   if (flows_free_.empty()) {
-    return nullptr;
+    f = new flow();
+    if (!f)
+      return nullptr;
```

Comment:
If allocation succeeded, the control flow should fall through.

Actually `new` won't return a nullptr but an exception upon a failure, so the if statement is not really necessary.